tests/animalShelter.test.ts

```diff
@@ -106,6 +106,21 @@ describe('Animal Shelter Tests', () => {
     expect(animals.mamamls).toBe(3)
     expect(animals.reptiles).toBe(2)
   })
+  test('reportOfNeededSpace should return the needed space for the specific setof animals', () => {
+    const animalShelter = AnimalShelter.getInstance()
```

Comment:
why not add a test with some reptiles? also a test with no animals ?  A test with child animals  could also be valuable